applovin_max/ios/Classes/AppLovinMAXAdViewUIComponent.m

```diff
@@ -0,0 +1,184 @@
+
+@property (nonatomic, strong) MAAdView *adView;
+@property (nonatomic, weak, nullable) AppLovinMAXAdView *containerView;
+
+@end
+
+@implementation AppLovinMAXAdViewUIComponent
+
+- (instancetype)initWithAdUnitIdentifier:(NSString *)adUnitIdentifier adFormat:(MAAdFormat *)adFormat
+{
+    self = [super init];
+    if ( self )
+    {
+        self.adView = [[MAAdView alloc] initWithAdUnitIdentifier: adUnitIdentifier adFormat: adFormat sdk: [AppLovinMAX shared].sdk];
+        self.adView.delegate = self;
+        self.adView.revenueDelegate = self;
+        
+        // Set this extra parameter to work around a SDK bug that ignores calls to stopAutoRefresh()
+        [self.adView setExtraParameterForKey: @"allow_pause_auto_refresh_immediately" value: @"true"];
+        
+        // Set a frame size to suppress an error of zero area for MAAdView
+        self.adView.frame = (CGRect) { CGPointZero, adFormat.size };
+    }
+    return self;
+}
+
+- (void)setPlacement:(NSString *)placement
+{
+    self.adView.placement = placement;
+}
+
+- (void)setCustomData:(NSString *)customData
+{
+    self.adView.customData = customData;
+}
+
+- (void)setAdaptiveBannerEnabled:(BOOL)adaptiveBannerEnabled
+{
+    [self.adView setExtraParameterForKey: @"adaptive_banner" value: adaptiveBannerEnabled ? @"true" : @"false"];
+}
+
+- (void)setAutoRefresh:(BOOL)autoRefresh
+{
+    if ( autoRefresh )
+    {
+        [self.adView startAutoRefresh];
+    }
+    else
+    {
+        [self.adView stopAutoRefresh];
+    }
+}
+
+-(void)setExtraParameters:(NSDictionary<NSString *, id> *)parameterDict
+{
+    for (NSString *key in parameterDict)
+    {
+        NSString *value = (NSString *) parameterDict[key];
+        [self.adView setExtraParameterForKey: key value: value];
+    }
+}
+
+-(void)setLocalExtraParameters:(NSDictionary<NSString *, id> *)parameterDict
+{
+    for (NSString *key in parameterDict)
+    {
+        id value = parameterDict[key];
+        [self.adView setLocalExtraParameterForKey: key value: value];
+    }
+}
+
+- (BOOL)hasContainerView
+{
+    return self.containerView != nil;
+}
+
+- (void)attachAdView:(AppLovinMAXAdView *)view
+{
+    self.containerView = view;
+}
+
+- (void)detachAdView
+{
+    self.containerView = nil;
+    
+    [self.adView removeFromSuperview];
+}
+
+- (void)loadAd
+{
+    [self.adView loadAd];
+}
+
+- (void)destroy
+{
+    [self detachAdView];
+    
+    self.adView.delegate = nil;
+    self.adView.revenueDelegate = nil;
+}
+
+#pragma mark - MAAdDelegate Protocol
+
+- (void)didLoadAd:(MAAd *)ad
+{
+    NSDictionary *adInfo = [[AppLovinMAX shared] adInfoForAd: ad];
+    
+    [[AppLovinMAX shared] sendEventWithName:@"OnNativeUIComponentAdViewAdLoadedEvent" body: adInfo];
+    
+    if ( self.containerView )
+    {
+        [self.containerView sendEventWithName: @"OnAdViewAdLoadedEvent" body: adInfo];
+    }
+    else
+    {
+        [self setAutoRefresh: NO];
```

Comment:
Is this necessary? If the `MAAdView` is not attached to a parent view, it should be smart enough not to auto-rfresh (feel free to test though..).

If we could remove this that would be great, since I'm concerned about managing the auto-refresh state and don't want to accidentally stop auto-refresh forever

Reply:
Removed the auto-refresh.